crates/matrix-sdk/src/client/mod.rs

```diff
@@ -830,6 +815,7 @@ impl Client {
     /// Similar to [`Client::add_event_handler`], but only allows functions
     /// or closures with exactly the three arguments [`Notification`],
     /// [`room::Room`], [`Client`] for now.
+    #[deprecated = "Use add_event_handler which also accepts functions of the same signature"]
```

Comment:
(Should we do a deprecation notice in the form of "this will be removed in version X"?)

Reply:
I don't think it buys us anything 🤷🏼

Reply:
What we do need to do though is update the docs of `add_event_handler()`, because it's not obvious that it can handle this use-case as well.

I'm even inclined to leave this method in and not deprecate it at all since  cramping `add_event_handler` with more docs and examples is particularly nice.

Reply:
Just delete the code™️ .